Formula/tcpreplay.rb

```diff
@@ -17,17 +17,36 @@ class Tcpreplay < Formula
   depends_on "libtool" => :build
   depends_on "libdnet"
 
+  uses_from_macos "libpcap"
+
   def install
-    ENV["MACOSX_DEPLOYMENT_TARGET"] = MacOS.version
     args = %W[
       --disable-debug
       --disable-dependency-tracking
       --disable-silent-rules
       --prefix=#{prefix}
       --enable-dynamic-link
-      --with-macosx-sdk=#{MacOS.version}
     ]
 
+    on_macos do
+      ENV["MACOSX_DEPLOYMENT_TARGET"] = MacOS.version
+      args << "--with-macosx-sdk=#{MacOS.version}"
+
+      # The SDK is currently found using `xcrun --sdk macosx<V>` starting with
+      # input `--with-macosx-sdk=<V>` and then going from older 10.8 onward.
+      # On ARM, for Big Sur 11.4 the correct SDK is 11.3 (as of 2021-07-11);
+      # however, the logic picks 10.15, which causes configure failure.
+      # As a workaround, we remove all 10.x versions from SDK detection logic.
+      #
+      # Check in next release if the workaround can be removed.
+      # Upstream issue: https://github.com/appneta/tcpreplay/issues/668
+      inreplace "configure.ac", /(\$with_macosx_sdk[ \t]+)(?:10\.\d+[ \t]+)+/, "\\1" if Hardware::CPU.arm?
```

Comment:
Another option would be to use `xcrun --show-sdk-version` and use the result in `--with-macosx-sdk=...`. I don't know if this is guaranteed to output.